pubtools/_pulp/tasks/fix_cves.py

```diff
@@ -0,0 +1,167 @@
+        erratum_page = self.get_erratum_from_pulp(advisory_id)
+        erratum = [ert for ert in erratum_page]
+        if not erratum:
+            self.fail("No erratum found for %s", advisory_id)
+        # multiple erratums should not exist
+        assert len(erratum) == 1
+
+        return erratum[0]
+
+    def get_erratum_from_pulp(self, advisory_id):
+        crit = Criteria.and_(
+            Criteria.with_unit_type(ErratumUnit), Criteria.with_id(advisory_id)
+        )
+        return self.pulp_client.search_content(criteria=crit).result()
+
+    @step("Process erratum")
+    def process_erratum_for_upload(self, erratum, cves):
+        # get non-cve erratum references
+        refs = [ref for ref in erratum.references if ref.type != "cve"]
+
+        # add new cves to erratum
+        for cve in cves:
+            refs.append(
+                ErratumReference(
+                    type="cve",
+                    title=cve,
+                    id=cve,
+                    href="https://www.redhat.com/security/data/cve/%s.html" % cve,
+                )
+            )
+
+        # updated version
+        version = bump_erratum_version(erratum)
+
+        # update erratum
+        # clear repository memberships. It will be evaluated by Pulp.
+        erratum = attr.evolve(
+            erratum, references=refs, repository_memberships=None, version=str(version)
+        )
+
+        LOG.debug("Processed erratum: %s", attr.asdict(erratum))
+
+        return erratum
+
+    @step("Upload erratum")
+    def upload_erratum(self, erratum, repos):
+        # find repo to upload
+        repo = self.get_upload_repo(repos)
+
+        # upload erratum to the repo
+        repo.upload_erratum(erratum).result()
+        LOG.info("Uploaded to %s", repo.id)
+
+    def get_repo_from_pulp(self, repo_id):
+        crit = Criteria.with_id(repo_id)
+        resp = self.pulp_client.search_repository(criteria=crit)
+
+        return [r for r in resp.result()]
+
+    def get_upload_repo(self, repos):
+        return random.choice(repos)
+
+    @step("Get affected repos")
+    def get_affected_repos(self, erratum):
+        # get repos listed in repository memberships
+        repo_ids = erratum.repository_memberships
+
+        repos = self.get_repo_from_pulp(repo_ids)
+        LOG.info("Affected repos: %s", ", ".join(list(repo_ids)))
+
+        return repos
+
+    @step("Compare CVEs")
+    def same_cves(self, erratum, cves):
+        # get cves from erratum
+        erratum_cves = [ref.id for ref in erratum.references if ref.type == "cve"]
+        LOG.info("CVEs from erratum: %s", ", ".join(sorted(erratum_cves)))
+
+        cves = [cve_id.upper() for cve_id in cves]
+        LOG.info("CVEs for fix: %s", ", ".join(sorted(cves)))
+
+        # compare with new cves
+        if set(erratum_cves) == set(cves):
+            LOG.info(
+                "New CVEs are same as the ones on the advisory. Nothing to update."
+            )
+            return 1
```

Comment:
Minor - How about returning booleans rather than using 1/0 for true/false?

Reply:
Done